tests/IntegrationTest.php

```diff
@@ -322,6 +322,10 @@ function ($e) use (&$wait) {
 
     public function testWaitingForSuccessfullyClosedConnectionShouldNotCreateAnyGarbageReferences()
     {
+        if (PHP_VERSION_ID < 70000) {
+            $this->markTestSkipped('Not support on PHP 5.x, due to GC internals');
+        }
```

Comment:
This is an interesting one. This seems to suggest this changeset introduces a cyclic reference for legacy PHP?

Reply:
Yes, during tests about 560ish cycles (depending on some fiddling) are left for the GC to clean up on PHP 5.x. PHP 7.x handles this without any issues. When running `gc_collect_cycles` directly after `HappyEyeBallsConnectionBuilder::connect` resolves but before return the promise will result in 0 cycles collect by the test `gc_collect_cycles`. In the end this is because of the design of the `HappyEyeBallsConnector` to use a `HappyEyeBallsConnectionBuilder` per connection attempt due to the state that has be to tracked for each connection attempt.

Reply:
So, I've disabled happy eyeballs by default on PHP 5.x because it could be to much of an impact and should be a conscious decision on legacy PHP. Since PHP 7 handles it just fine it's enabled by default there.